config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
```

Comment:
Re-reviewed your Raft paper and I see why this is dangerous. After an outage we have to do the best we can because we don't have the right peers to determine if the last few logs were committed or not, so the best we can do is assume they were. We wouldn't want to do this process any other time.

I don't think it's going to be safe to ingest the peers.json initially like I'm doing now, though. It would be much better to just delete it.